src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -632,6 +651,21 @@ public virtual void remove_rollback_directory_if_exists(string packageName)
                 uninstallSuccessAction: null,
                 addUninstallHandler: false);
 
+            EventHandler<PackageOperationEventArgs> beforeModifyHandler = (sender, eventArgs) =>
+            {
+                // Soundness: any code called here must avoid calling packageManager methods for
+                // InstallPackage() and UninstallPackage(), or in some other way handle the

```

Comment:
There's a few places that we have this comment, is there anything we can do to check this doesn't happen?

Reply:
I'm open to ideas but I'm not sure there's a great way to ensure this. Also, in at least one case (though in code I haven't touched here) I think we actively and knowingly do call UninstallPackage from the PackageUninstalled(/ing?) handler, though at least there we do guard against the recursion by checking how many times it's been called.

So... like yeah it'd be good to have, but I don't know that there's a good way to go about it.

Reply:
The only way I could think to maybe do it would be looking at the call stack or something like that, and not sure it'd be worth it in most instances due to performance? Maybe some unit tests that does it?

I'll leave this conversation open for @gep13 to weigh in on if he has any opinions or ideas about it.

Reply:
I'm not sure if you can get _that_ kind of funky in C# land even with Reflection, but if you can I don't think it's a good idea to do it here 😂

Reply:
To prevent usage of these methods, we could add them to the BannedAPI's list, and provide an exemption for the current uses of it.  That way, any new usage of it will be flagged, and will need to be overridden.  The only slight issue here is that I believe we have only introduced the BannedAPI's on the develop branch, so may not be able to use it here.